### PR TITLE
All Product Reviews: Tests

### LINKED ISSUE #1392: Product Reviews: Add tests
Parent ticket: #1324 

Once all the other functionality has been added, remove unused instrumentation and unit test and add new ones that work with the new designs/architecture (MVVM)

### PR DESCRIPTION
**Pull request by AmandaRiu**
This PR fixes #1392 by adding tests for the `Reviews` component. I was able to add a bunch of unit tests for the ViewModel, and I fixed the `MainNavigatorTest` for the Reviews tab by adding all the Dagger dependencies. I didn't get to add as many tests as I would've liked because there are a lot of broken tests in the instrumentation tests section. I spent a bunch of time trying to iron them all out, but I'll need to wait until full tests are added for Refunds and the new v4 stats code. For now the tests in place are a very good baseline to start with. I dont wnat to hold up getting this feature to production this sprint, so I'm submitting this as-is. 

To Test:

- Run the Unit tests `./gradlew testWasabiDebugUnitTest`
- Run the tests in the `MainNavigatorTests`

All tests should pass

### CODE REVIEW

File: WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/main/MainNavigationTest.kt [hunk 1]
```diff
@@ -47,9 +47,15 @@ class MainNavigationTest : TestBase() {
...
+        // Make sure the bottom navigation view is showing
+        activityTestRule.activity.showBottomNav()
+
+        // add mock data to order list screen
+        activityTestRule.setOrderListWithMockData()
+
         // Verify switching bottom bar tabs does not retain a back stack.
         // Switch from the default dashboard tab to the reviews tab
-        onView(withId(R.id.reviews)).perform(click())
+        onView(withId(R.id.orders)).perform(click())
```
Comment (nbradbury) [inline WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/main/MainNavigationTest.kt]: The comment above this states it's switching to the reviews tab, but instead it's switching to the orders tab.

File: WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/main/MainNavigationTest.kt [hunk 2]
```diff
@@ -91,25 +97,36 @@ class MainNavigationTest : TestBase() {
...
         activityTestRule.activity.showBottomNav()
 
+        // add mock data to the reviews list screen
+        activityTestRule.setReviewListWithMockData()
+
         // Select the reviews bottom bar option
         onView(withId(R.id.reviews)).perform(click())
 
         // Verify the toolbar title has changed to Reviews
         onView(withId(R.id.toolbar)).check(matches(
```
Comment (nbradbury) [inline WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/main/MainNavigationTest.kt]: There are two checks here for the same toolbar title (it's the same string with different resource names).

File: WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/main/MainNavigationTest.kt [hunk 3]
```diff
@@ -91,25 +97,36 @@ class MainNavigationTest : TestBase() {
...
                 WCMatchers.withToolbarTitle(equalToIgnoringCase(appContext.getString(R.string.review_notifications)))))
+
+        // Verify the toolbar title has changed to 'Reviews'
```
Comment (nbradbury) [inline WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/main/MainNavigationTest.kt]: There are two checks here for the same toolbar title (it's the same string with different resource names).

**Comment by nbradbury** [comment]
Great work on this, @AmandaRiu. Just two minor nits, then I'm good to merge.

File: WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/main/MainNavigationTest.kt (hunk 1 above)
Comment (AmandaRiu) [reply WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/main/MainNavigationTest.kt]: Nice catch! Fixed adf494d74636b26a9e137f254ef01dd62127fb2e

File: WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/main/MainNavigationTest.kt (hunk 3 above)
Comment (AmandaRiu) [reply WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/main/MainNavigationTest.kt]: Fixed in adf494d74636b26a9e137f254ef01dd62127fb2e

**Comment by AmandaRiu** [comment]
Thanks for the review @nbradbury ! I've addressed all comments. Ready for another round!

**Comment by nbradbury** [comment]
:shipit:

**Pull request merged**